src/arkreactor/Compiler/AST/makeErrorCtx.cpp

```diff
@@ -11,30 +12,90 @@ namespace Ark::internal
 {
     void makeContext(std::ostream& os, const std::string& code, std::size_t line, std::size_t col_start, std::size_t sym_size)
     {
+        os << termcolor::colorize;
         std::vector<std::string> ctx = Utils::splitString(code, '\n');
 
-        std::size_t col_end = std::min(col_start + sym_size, ctx[line].size());
+        std::size_t col_end = std::min<std::size_t>(col_start + sym_size, ctx[line].size());
         std::size_t first = line >= 3 ? line - 3 : 0;
         std::size_t last = (line + 3) <= ctx.size() ? line + 3 : ctx.size();
+        LineColorContextCounts line_color_context_counts;
 
         for (std::size_t loop = first; loop < last; ++loop)
         {
-            os << std::setw(5) << (loop + 1) << " | " << ctx[loop] << "\n";
+            std::string current_line = colorizeLine(ctx[loop], line_color_context_counts);
+            os << termcolor::green << std::setw(5) << (loop + 1) << termcolor::reset << " | " << current_line << "\n";
 
             if (loop == line)
             {
                 os << "      | ";
 
-                for (std::size_t j = 0; (sym_size > col_start) ? false : (j < col_start); ++j)
+                // padding of spaces
+                for (std::size_t i = 0; i < col_start; ++i)
+                    os << " ";
+
+                // underline the error
+                os << termcolor::red;
+                for (std::size_t i = col_start; i < col_end; ++i)
+                    os << "^";
+
+                os << termcolor::reset << "\n";
+            }
+        }
+    }
+
+    std::string colorizeLine(const std::string& line, LineColorContextCounts& line_color_context_counts)
+    {
+        std::vector<std::ostream& (*)(std::ostream & stream)> pairing_color {
+            termcolor::bright_blue,
+            termcolor::bright_green,
+            termcolor::bright_yellow
+        };
+        std::size_t pairing_color_size = pairing_color.size();
```

Comment:
Because this isn't meant to change each time the function is called, you might want to make that constexpr and use an std::array.

Reply:
When I was first writing this, I was trying to achieve something like this by making the vector static and eventually gave up. :laughing:

This is a really good idea idea. I've made that change. Initially, I was trying to avoid using std::array just because I would have to pass in the size of the array as a template parameter, but I think this is better now.